### PR TITLE
feat: post deploy checks

### PR DESCRIPTION
**Pull request by deanamiel**
ITS post deploy checks for config values as per [AXE-2774](https://axelarnetwork.atlassian.net/jira/software/c/projects/AXE/boards/21?selectedIssue=AXE-2774)

[AXE-2774]: https://axelarnetwork.atlassian.net/browse/AXE-2774?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: evm/its.js [hunk 1]
```diff
@@ -437,6 +463,36 @@ async function processCommand(_, chain, options) {
             break;
         }
 
+        case 'postDeploy': {
```
Comment (milapsheth) [inline evm/its.js:466]: should just call it checks since you can run it whenever (even post upgrades)

File: evm/its.js [hunk 2]
```diff
@@ -437,6 +463,36 @@ async function processCommand(_, chain, options) {
...
+
+            const toCheck = {
+                tokenManagerDeployer: await interchainTokenService.tokenManagerDeployer(),
+                interchainTokenDeployer,
+                interchainToken,
+                tokenManager: await interchainTokenService.tokenManager(),
+                tokenHandler: await interchainTokenService.tokenHandler(),
+                implementation: await interchainTokenService.implementation(),
+                interchainTokenFactory,
+                interchainTokenFactoryImplementation,
```
Comment (milapsheth) [inline evm/its.js:488]: should also check all constructor args used for deployment, and at least print those for manual verification. We get the constructor args in verify-contract script so you can create a helper maybe.

Comment (deanamiel) [reply evm/its.js:554]: added some additional checks

File: evm/its.js [hunk 3]
```diff
@@ -79,7 +84,42 @@ const decodeMulticallData = async (encodedData, contractJSON) => {
     });
 };
 
-async function processCommand(_, chain, options) {
+function compare(contractValue, configValue, variableName) {
+    contractValue = isNonEmptyString(contractValue) ? contractValue.toLowerCase() : contractValue;
+    configValue = isNonEmptyString(configValue) ? configValue.toLowerCase() : configValue;
+
+    if (contractValue === configValue) {
+        printInfo(`Confirmed: Value match for '${variableName}'.`);
```
Comment (milapsheth) [inline evm/its.js:92]:
```suggestion
        printInfo(`${variableName}`, contractValue);
```
normal print can be considered success

File: evm/its.js [hunk 4]
```diff
@@ -79,7 +84,42 @@ const decodeMulticallData = async (encodedData, contractJSON) => {
...
+        );
+    }
+}
+
+function configCheck(contractName, contractConfig, toCheck) {
+    let allKeysMatch = true;
+
+    for (const [key, value] of Object.entries(toCheck)) {
+        if (Object.prototype.hasOwnProperty.call(contractConfig, key)) {
+            if (contractConfig[key] === value) {
```
Comment (milapsheth) [inline evm/its.js:105]: duplicate code, should reuse `compare`

File: evm/verify-contract.js [hunk 5]
```diff
@@ -13,6 +13,17 @@ const { Command, Option } = require('commander');
 const { verifyContract, getEVMAddresses, printInfo, printError, mainProcessor, getContractJSON } = require('./utils');
 const { addBaseOptions } = require('./cli-utils');
 
+async function getTrustedChainsAndAddresses(config, interchainTokenService) {
```
Comment (milapsheth) [inline evm/verify-contract.js:16]: moved this to `its.js` since we're defining its utils there

File: evm/its.js [hunk 6]
```diff
@@ -471,6 +511,54 @@ async function processCommand(_, chain, options) {
...
+
+            const toCheck = {
+                tokenManagerDeployer: await interchainTokenService.tokenManagerDeployer(),
+                interchainTokenDeployer,
+                interchainToken,
+                tokenManager: await interchainTokenService.tokenManager(),
+                tokenHandler: await interchainTokenService.tokenHandler(),
+                implementation: await interchainTokenService.implementation(),
+                interchainTokenFactory,
+                interchainTokenFactoryImplementation,
```
Comment (milapsheth) [inline evm/its.js:554]: can just loop and use compare here directly?

Comment (deanamiel) [reply evm/its.js:554]: removed `configCheck` to compare directly here

File: evm/its.js [hunk 7]
```diff
@@ -437,6 +466,61 @@ async function processCommand(_, chain, options) {
...
+            };
+
+            for (const [key, value] of Object.entries(toCheck)) {
+                if (Object.prototype.hasOwnProperty.call(contractConfig, key)) {
```
Comment (milapsheth) [inline evm/its.js:513]:
```suggestion
                if (contractConfig[key]) {
```
isn't this sufficient?

**Review by milapsheth — APPROVED** [review]

**Pull request merged**